WordPress/Classes/Utility/BlogQuery.swift

```diff
@@ -21,7 +21,7 @@ struct BlogQuery {
     }
 
     func username(_ username: String) -> Self {
-        and(NSPredicate(format: "account.username = %@", username))
+        and(NSPredicate(format: "account.username = %@ OR username = %@", username, username))
```

Comment:
I'm guessing this edit came because of the Objective-C code using `username =` above.

Is there any chance this could backfire on us? A usage of this method in the codebase where we really wanted to check only `account.username` and that is not covered by unit tests?

I don't see why that would be the case, but I thought I'd ask anyway since you're more familiar with the usages of these queries.

Reply:
In theory, yes, it's possible. But in practice, all places that use this username filter have other filters too (i.e. filter by site id for WP.com sites, or xmlrpc address for self hosted sites). I don't imagine that filtering _only_ by username would be useful to a feature.

But, I can see how this filter might be confusing, since we do have two difference kinds of "username"s after all, one for self hosted sites and the other for WP.com sites. I think it's best to make it clearer by splitting this filter into two — https://github.com/wordpress-mobile/WordPress-iOS/pull/19925/commits/f1571c3345cb1535a421995691edd24fdaff9d00